webserver.sh

```diff
@@ -69,13 +69,19 @@ EOF
 fi
 
 # spinning up an ubuntu vm
-if ( multipass info relativepath | grep Running )
-then 
-  echo "relativepath vm is running"
-else 
-  echo "launching a ubuntu vm named relativepath"
-  multipass launch --name relativepath --cloud-init cloud-init.yaml
+if ( multipass info relativepath )
+then
+  if ( multipass info relativepath | grep Running )
```

Comment:
Is there ever a way for the vm to be in a state other than deleted or Running? Are you handling a state that may not actually happen or can you demonstrate that there is a state we are not accounting for?

Reply:
We should start to talk about this script in context with the destroy script as a whole flow instead of individually. If we focus on the overall workflow of what we are trying to do -> build a vm, destroy a vm. Then we can adjust the script(s) based on what is the best way to support that flow.

Reply:
I will do some research on that and get back to you. For now it does work on a 'Stopped' state.

Reply:
Right, but how do we end up in a "Stopped" state?

Reply:
I don't believe a switching to a stopped state is explicitly executed in our codebase so far. If that's true and the main branch works with a stopped state, do we need to check for it? Are there any edge cases we aren't thinking of?

Reply:
works on a 'Suspended' state too.

Reply:
I've done some digging and maybe I need some help with this. I could only find 3 states of an existing vm. 'Running', 'Stopped' and 'Suspended'. The code I wrote gets a 'Stopped' and 'Suspended' state vm to a 'Running' state. I believe the code is a good addition and covers the states I could find.

Reply:
My question was more around asking if we even get to those states and if not let's not support them. The less code the better. So if we don't explicitly get to the stopped state in any of our code, we don't need to account for it, or any other state that we don't explicitly control getting to. So for what we're doing the only states we have are Running or doesn't exist.

Reply:
Team has agreed to close as not needed.